shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/compiler/ExpressionNode.java

```diff
@@ -309,7 +207,144 @@ private static Parser parse9(Parser input, Consumer<ExpressionNode> output) {
+                          .whitespace()
+                          .list(
+                              cases::set,
+                              (cp, co) -> {
+                                final AtomicReference<ExpressionNode> condition =
+                                    new AtomicReference<>();
+                                final AtomicReference<ExpressionNode> value =
+                                    new AtomicReference<>();
+                                final Parser cresult =
+                                    parse(
+                                        parse(
+                                                cp.whitespace().keyword("When").whitespace(),
+                                                condition::set)
+                                            .whitespace()
+                                            .keyword("Then")
+                                            .whitespace(),
+                                        value::set);
+                                if (cresult.isGood()) {
+                                  co.accept(new Pair<>(condition.get(), value.get()));
+                                }
+                                return cresult;
+                              })
+                          .whitespace()
+                          .keyword("Else")
+                          .whitespace(),
+                      alternative::set)
+                  .whitespace();
+          if (result.isGood()) {
+            o.accept(
+                new ExpressionNodeSwitch(
+                    p.line(), p.column(), test.get(), cases.get(), alternative.get()));
+          }
+          return result;
+        });
+    OUTER.addKeyword(
+        "For",
+        (p, o) -> {
+          final AtomicReference<DestructuredArgumentNode> name = new AtomicReference<>();
+          final AtomicReference<SourceNode> source = new AtomicReference<>();
+          final AtomicReference<List<ListNode>> transforms = new AtomicReference<>();
+          final AtomicReference<CollectNode> collector = new AtomicReference<>();
+          final Parser result =
+              p.whitespace()
+                  .then(DestructuredArgumentNode::parse, name::set)
+                  .whitespace()
+                  .then(SourceNode::parse, source::set)
+                  .whitespace()
+                  .symbol(":")
+                  .whitespace()
+                  .list(transforms::set, ListNode::parse)
+                  .then(CollectNode::parse, collector::set)
+                  .whitespace();
+          if (result.isGood()) {
+            o.accept(
+                new ExpressionNodeFor(
+                    p.line(),
+                    p.column(),
+                    name.get(),
+                    source.get(),
+                    transforms.get(),
+                    collector.get()));
+          }
+          return result;
+        });
+    OUTER.addRaw(
+        "tenary if",
```

Comment:
`tenary if`? (Probably doesn't need to be changed if this string is only exposed to/consumed by Shesmu, since it'll be deleted later)

Reply:
Fixed. It will appear in error messages.